dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpServlet.java

```diff
@@ -157,6 +172,16 @@ private ToolResponse toErrorResponse(Throwable t) {
         return ToolResponse.error(msg);
     }
 
+    private void checkAndParseCancellationObject(Object[] argumentsArray, RequestId requestId) {
+        for (Object argument : argumentsArray) {
+            if (argument instanceof Cancellation cancellation) {
+                ((CancellationImpl) cancellation).setRequestId(requestId);
+                connection.registerOngoingRequest(requestId, cancellation);
+                break;
+            }
+        }
+    }
```

Comment:
Not a fan of doing reflection to find the cancellation object like this. We should already know which elements contain a `Cancellation` from the tool metadata.

Let's leave it for now, but we may want to rework this in a follow-up.